projects/ui/src/Components/Apps/Details/AuthenticationSection/AppAuthenticationSection.tsx

```diff
@@ -1,35 +1,136 @@
-            <span>Client ID:</span>
-            <DetailsPageStyles.OAuthClientId>
-              {app.idpClientId}
-            </DetailsPageStyles.OAuthClientId>
-            {/*
-            // Designs show this "hidden" field, but we have the value.
-            // TODO: Figure out what to show here.
-            <DataPairPill pairKey={"Client Secret"} value={"hidden"} />
-            */}
-            <DataPairPill
-              pairKey={"Client Secret"}
-              value={app.idpClientSecret}
-            />
-          </DetailsPageStyles.OAuthClientRow>
+          {!!oauthError ? (
+            <DetailsPageStyles.OAuthClientRow mt="10px">
+              <Button onClick={() => setShowConfirmCreateCredentials(true)}>
+                Create OAuth Client
+              </Button>
+            </DetailsPageStyles.OAuthClientRow>
+          ) : credentialsToShow === undefined ? (
+            <DetailsPageStyles.OAuthClientRow>
+              <Flex sx={{ width: "100%" }} justify={"center"}>
+                <Loading />
+              </Flex>
+            </DetailsPageStyles.OAuthClientRow>
+          ) : (
+            <>
+              <DetailsPageStyles.OAuthClientRow>
+                <span>Client ID:</span>
+                <DetailsPageStyles.OAuthClientId>
+                  {credentialsToShow.idpClientId}
+                </DetailsPageStyles.OAuthClientId>
+                <DataPairPill pairKey={"Client Secret"} value={"hidden"} />
+              </DetailsPageStyles.OAuthClientRow>
+              <DetailsPageStyles.OAuthClientRow mt="10px">
+                <Button
+                  variant="outline"
+                  color="danger"
+                  onClick={() => setCredentialIdToDelete(credentialsToShow.id)}
+                >
+                  Delete OAuth Client
+                </Button>
+              </DetailsPageStyles.OAuthClientRow>
+            </>
+          )}
         </Box>
       </GridCardStyles.GridCard>
+      {/*
+
+      // region Modals
+      */}
+      <ConfirmCreateOAuthModal
+        appId={app.id}
+        onClose={() => setShowConfirmCreateCredentials(false)}
+        onCreatedClient={(newCredentials) => {
+          setCredentialsToShow(newCredentials);
+        }}
+        open={showConfirmCreateCredentials}
+      />
+      <ConfirmDeleteOAuthModal
+        credentialId={credentialIdToDelete}
+        appId={app.id}
+        onClose={() => setCredentialIdToDelete("")}
+        open={credentialIdToDelete !== ""}
+      />
+      <ViewCreatedItemModal
+        createdObjectName="OAuth Client"
+        itemToCopyName="Client Secret"
+        itemToCopyValue={credentialsToShow?.idpClientSecret ?? ""}
+        open={!!credentialsToShow?.idpClientSecret}
+        onCloseModal={() => {
+          if (credentialsToShow) {
+            setCredentialsToShow({
+              ...credentialsToShow,
+              idpClientSecret: "",
+            });
+          }
+        }}
+        additionalContentTop={
```

Comment:
I'm a bit surprised this isn't more standardized.  This isn't really a 1-off is it?

Reply:
The `ViewCreatedItemModal` is being used in the Authorization section and the API Keys section. I wanted to not have 2 components doing the same thing, so that's why I added this in and made it generic.